### DISCUSSION TITLE
Open file/line of code in Github

### DISCUSSION THREAD

**Discussion by christiand** [zed-industries/zed#12104] [Feature Requests]
There are a few discussions/issues open about git & github integration, but an essential feature I use many times per day in IntelliJ is to click on a line of code, and have it open the github URL with highlighted line of code (or put it right in the clipboard).

Then I can share a code pointer to a colleague. or dump in a Jira ticket, paste into slack, etc.

**Comment by CharlesChen0823** [comment]
might in `ctrl-shift-p` with `copy permalink link to line`

**Comment by SomeoneToIgnore** [comment]
There's an `editor: open permalink to line` command present (`cmd-shift-p` opens the command palette on mac).

Also, if `editor: toggle git blame` command toggled on the names and commits on the left of the editor, clicking any item there would also open a corresponding commit on GitHub.

I do not think that we will add anything else on top that will 

> click on a line of code, and have it open the github URL

as clicking on a line of code is already somewhat occupied by cmd-click to open various URLs and definitions.

**Comment by christiand** [comment]
ah that's perfect. thank you!

**Comment by jaydenseric** [comment]
Note that this functionality is broken in some cases, see:

https://github.com/zed-industries/zed/issues/11160#issuecomment-2125955332

**Comment by dabeeeenster** [comment]
It would be great to expose this in a right click menu - i had to search to find this functionality on github
